imsv-docs-astro/src/content/docs/guides/authentication/authentication.mdoc

```diff
@@ -0,0 +1,192 @@
+page="resources/public-sandbox-account" title="sandbox credentials" /%} to
+quickly get started. [Contact us](https://immersve.com/contact/) if you are
+ready to provision credentials for your application.
+
+
+## API Key Authentication
+
+The API Key authentication scheme provides enduring access to Immersve APIs. API
+Key authentication is most suitable when your client integration is running on
+your backend server. This will typically be the case for a {% link
+page="use-cases/custodial-wallets" /%} card issuing solution.
+
+Requests using API Key authentication include the following HTTP headers:
+
+{% table %}
+* Header
+* Description
+---
+* <nobr>x-api-key</nobr>
+* Identifier for the Immersve API key. May be logged.
+---
+* <nobr>x-api-secret</nobr>
+* Secret for the Immersve API key. Should not be logged.
+---
+* <nobr>x-account-id</nobr>
+* Optional target account hint for permission resolution. Defaults to the
+  account that owns the API key.
+{% /table %}
+
+The permissions granted to an API Key are determined by its role. See the Roles
+and Permissions section below for the available roles.
+
+
+## User Session Authentication
+
+The User Session authentication scheme allows client applications to act on
+behalf of users within scopes explicitly granted by the user. User Session
+authentication is most suitable when your client integration is running directly
+on the user's device. This will typically be the case for a {% link
+page="use-cases/non-custodial-wallets" /%} card issuing solution.
+
+Requests using User Session authentication include the following HTTP headers:
+
+{% table %}
+* Header
+* Description
+---
+* authorization
+* The access token obtained via user login or token refresh. The value must be
+  prefixed with "Bearer ".
+---
+* <nobr>x-account-id</nobr>
+* Optional target account hint for permission resolution. Defaults to the
+  primary account associated with the authenticated session.
+{% /table %}
+
+User Session tokens are created by using the {% link endpoint="initiate-login"
+/%} and {% link endpoint="complete-login" /%} endpoints. After completing
+login, you will have an access token and a refresh token.
+The access token will be used for subsequent interactions with Immersve API
+endpoints. The refresh token will be used to renew an expired access token.
+
+Immersve users are identified by the web3 address used to login. Login methods
+for each of the {% link page="guides/supported-chains" /%} are defined in the Login
+section below.
+
+The permissions granted to an access token are determined by the role of the
+authenticated user (with respect to the target account) along with the scope
+requested by the client application when initiating login. Available roles and
+scopes are described in the sections below.
+
+
+## Roles and Permissions
+
+Roles define the permissions available to a User or API Key. The
+following roles are available.
```

Comment:
This gives the impression that the roles are input they must choose.

```suggestion
Roles define the permissions available to a User or API Key. API keys are issued with the following roles.
```

Reply:
Also the integration would have no knowledge of roles when performing User Session Authentication. 
Not sure how to make that clear. You don specify the role granted to users through self-custodial flow...

Reply:
This is already explicitly mentioned in the description for the account-owner role.

Reply:
Roles are designed to be explicitly chosen. We just don't have those APIs yet.